lib/core/src/Cardano/Wallet.hs

```diff
@@ -1545,14 +1544,40 @@ balanceTransaction
     => ctx
     -> ArgGenChange s
     -> (W.ProtocolParameters, Cardano.ProtocolParameters)
-    -> TimeInterpreter (Either PastHorizonException)
+    -- ^ 'Cardano.ProtocolParameters' can be retrived via a Local State Query to
+    -- a local node.
+    --
+    -- Providing incorrect values will result in a phase 1 script integrity
+    -- hash failures instead of phase 2 failures if script redeemers are
+    -- present in the transaction, ensuring collateral is never forfeited.
+    --
+    -- TODO: Remove 'W.ProtocolParameters' argument.
+    -> (Cardano.EraHistory Cardano.CardanoMode, SystemStart)
```

Comment:
The `SystemStart` _is_ duplicated with `getGenesisBlockDate` in `W.ProtocolParameters` which we'd like to replace, but _not_ `Cardano.ProtocolParameters`.